### PR TITLE
Refactor \Order\Shipment\AddTrack Controller to use ResultInterface

### PR DESCRIPTION
**Pull request by Jeroenll**
### Description (*)
Refactor \Order\Shipment\AddTrack Controller to use ResultInterface and Repository to save shipment.

### Fixed Issues (if relevant)
1. None I could find.

### Manual testing scenarios (*)
1. Place order.
2. Create shipment without tracking information.
3. Add tracking information to shipment.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @JeroenVanLeusden. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Comment by Jeroenll** [comment]
Will update the tests if necessary when implementation itself is correct.

**Review by VladimirZaets — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @VladimirZaets, thank you for the review.
[ENGCOM-4532](https://jira.corp.magento.com/browse/ENGCOM-4532) has been created to process this Pull Request

**Comment by soleksii** [comment]
:heavy_check_mark: QA Passed

File: app/code/Magento/Shipping/Controller/Adminhtml/Order/Shipment/AddTrack.php [hunk 1]
```diff
@@ -18,56 +31,84 @@ class AddTrack extends \Magento\Backend\App\Action
...
     /**
-     * Add new tracking number action
+     * Add new tracking number action.
      *
-     * @return void
-     * @throws \Magento\Framework\Exception\LocalizedException
+     * @return ResultInterface
      */
-    public function execute()
+    public function execute(): ResultInterface
```
Comment (sidolov) [inline app/code/Magento/Shipping/Controller/Adminhtml/Order/Shipment/AddTrack.php]: It's a good idea to add strict type to the method but unfortunately, we are not allowed to change public methods signature according to our [Backward Compatible Development Guide](https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/)

**Review by sidolov — CHANGES_REQUESTED** [review]

File: app/code/Magento/Shipping/Controller/Adminhtml/Order/Shipment/AddTrack.php (hunk 1 above)
Comment (Jeroenll) [reply app/code/Magento/Shipping/Controller/Adminhtml/Order/Shipment/AddTrack.php]: @sidolov Removed the return type.

**Review by sidolov — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sidolov, thank you for the review.
[ENGCOM-4532](https://jira.corp.magento.com/browse/ENGCOM-4532) has been created to process this Pull Request

**Comment by soleksii** [comment]
:heavy_check_mark: QA Passed

**Pull request merged**